BITSWAP.md

```diff
@@ -1,200 +1,200 @@
+bytes, encoded as an unsigned variable length integer as defined by the
+[multiformats unsigned-varint spec][uvarint-spec].
 
-## Message Queue
+All protocol messages must be less than or equal to 4MiB in size
 
-Each active peer has an associated message queue. The message queue holds the next message to be sent to that peer. The message queues receive updates from two other subsystems:
+### Block Sizes
 
-1.  Wantlist manager: When a `CID` is added or removed from our wantlist, we must update our peers – these wantlist updates are sent to all relevant peers’ message queues.
-2.  Decision engine: When we have a block that a peer wants and the decision engine decides to send the block, we propagate the block to that peer’s message queue.
+Bitswap implementations must support sending and receiving individual blocks of sizes up to 2MiB
 
-Task workers watch the message queues, dequeue a waiting message, and send it to the recipient.
+## Bitswap 1.1.0
 
-## Network
+Bitswap 1.1.0 introduces a 'payload' field to the protobuf message and deprecates the
+existing 'blocks' field. The 'payload' field is an array of pairs of cid
+prefixes and block data. The cid prefixes are used to ensure the correct
+codecs and hash functions are used to handle the block on the receiving
+end.
 
-The network is the abstraction representing all Bitswap peers that are connected to us by one or more hops. Bitswap messages flow in and out of the network. This is where a game-theoretical analysis of Bitswap becomes relevant – in an arbitrary network we must assume that all of our peers are rational and self-interested, and we act accordingly. Work along these lines can be found in the [research-bitswap repository](https://github.com/ipfs/research-bitswap), with a preliminary game-theoretical analysis currently in-progress [here](https://github.com/ipfs/research-bitswap/blob/docs/strategy_analysis/analysis/prelim_strategy_analysis.pdf).
+It is otherwise identical to 1.0.0
 
-# Implementation Details
+### Wire Format
 
-## Coalescing Messages
-
-When a message queue that already contains a Bitswap message receives another, the new message should be coalesced with the original to reduce the overhead of sending separate packets.
-
-## Bitswap Sessions
-
-Bitswap sessions are an attempt to optimize the block requests sent to other Bitswap clients. When requesting a graph of blocks from the network, we send a wantlist update containing the graph’s root block to all of our peers. Then, for each peer who sends the root block back, we add that peer to the graph’s *active set*. We then send all requests for other nodes in the graph only to the peers in the active set. The idea is that peers who have the root node of a graph are likely to have its children as well, while those who do not have the root are unlikely to have its children.
+```
+message Message {
 
-**TODO**: Everything below must either be updated/integrated above, or removed
+  message Wantlist {
```

Comment:
This seems to have some of the 1.2 fields. It should be fixed to correctly represent 1.1

Reply:
fixed